modules/statemachine/statemachinemodule.cpp

```diff
@@ -116,6 +118,23 @@ bool StateMachineModule::canGoToState(const std::string& state) const {
     return _machine->canTransitionTo(state);
 }
 
+void StateMachineModule::saveToFile(const std::string& filename,
+                                    std::optional<std::string> directory) const
```

Comment:
In this case it might tbe better to say:
```
void StateMachineModule::saveToFile(const std::string& filename, std::string directory = "${TEMPORARY}/")
```
in the header

Then you don't need to worry about the whole optional thingy

Reply:
That's a very good point you make there ,sir 😁